test/compress/reduce_vars.js

```diff
@@ -4819,3 +4819,54 @@ issue_2485: {
     }
     expect_stdout: "6"
 }
+
+issue_2496: {
+    options = {
+        passes: 2,
+        reduce_funcs: true,
+        reduce_vars: true,
+        toplevel: true,
+        unused: true,
+    }
+    input: {
+        function execute(callback) {
+            callback();
+        }
+        class Foo {
+            constructor(message) {
+                this.message = message;
+            }
+            go(value) {
+                this.message = "PASS";
+                console.log(this.message);
+            }
+            run() {
+                execute(() => {
+                    this.go(3);
+                });
+            }
+        }
+        new Foo("FAIL").run();
+    }
+    expect: {
+        class Foo {
+            constructor(message) {
+                this.message = message;
+            }
+            go(value) {
+                this.message = "PASS";
+                console.log(this.message);
+            }
+            run() {
+                (function(callback) {
+                    callback();
+                })(() => {
+                    this.go(3);
```

Comment:
The argument `3` to `go(value)` are unused remnants from reducing a larger test case. You could drop the `3` and `value` as superfluous.

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2504/commits/35d116b517e878e32ddcab9bada137e76cc2f155